docs/docs/porting-create-react-app-to-gatsby.md

```diff
@@ -0,0 +1,263 @@
+
+Understanding the distinction between the client (or browser) and server will help you understand that key difference between Create React App and Gatsby. Create React App doesn't by default render your components with server-side rendering APIs when it is built like Gatsby does.
+
+_Check out Dustin Schau's [blog post about Gatsby internals](/blog/2019-04-02-behind-the-scenes-what-makes-gatsby-great/#server-side-rendering-ssr-at-build-time) that explains the technical aspects of the build process in greater detail_
+
+The `gatsby build` command won't be able to use browser APIs so some code would cause your build to break if it isn't protected.
+
+Some common APIs that would need to be protected are:
+
+- `window`
+- `localStorage`
+- `navigator`
+- packages like `react-router-dom`
+
+These are only a few examples, though all can be fixed in one of two ways:
+
+1. wrapping the code in an `if` to check if whatever you are referencing is defined so builds won't try to reference something undefined and the browser will be able to deal with it fine:
+
+```jsx
+if (typeof window !== `undefined`) {
+  // code using window like window.location...
+}
+```
+
+2. moving references to them into a `componentDidMount` or `useEffect` hook:
+
+```jsx
+import React from "react"
+
+const Foo = () => {
+  window.alert("This will break the build")
+  return <span>Bar</span>
+}
+
+export default Foo
+```
+
+Would be changed to:
+
+```jsx
+import React from "react"
+
+const Foo = () => {
+  React.useEffect(() => {
+    window.alert("This won't break the build")
+  })
+  return <span>Bar</span>
+}
+
+export default Foo
+```
+
+For more information about errors encountered during builds, see the doc on [debugging HTML builds](/docs/debugging-html-builds/)
+
+### Routing
+
+There are two possibilites of routes that you can setup: static and dynamic. A static route has content that doesn't change, whereas a dynamic route can [fetch data at runtime](/docs/client-data-fetching/) just like any other React app.
+
+Gatsby automatically turns React components in the pages folder into static routes.
+
+> **Note**: An advantage to having pages in separate files like this is a defined way of [automatically code splitting](/docs/how-code-splitting-works/), whereas Create React App requires you to use the `import()` syntax to assign what elements should be loaded dynamically
+
+For dynamic routes, you should implement routing with [@reach/router](https://reach.tech/router), which is already included with Gatsby. Dynamic routes can be implemented the same way you would implement a router in Create React App (or any other React application). However, because these routes won't be represented as HTML files in the final build, if you want users to be able to visit the routes directly (like entering the URL in the search bar), you'll need to generate pages in the `gatsby-node.js` file which is demonstrated in the [Building Apps with Gatsby](/docs/building-apps-with-gatsby/) guide.
```

Comment:
Note: we'll want to do a big find and replace when we upgrade to React Router in v3.